torchft/process_group_test.py

```diff
@@ -60,24 +61,46 @@ def dummy_init_pg() -> None:
         )
 
 
-def _test_pg(
-    pg: ProcessGroup,
-    example_tensor: torch.Tensor = torch.randn((2, 3), dtype=torch.float32),
-) -> Dict[str, dist._Work]:
+# pyre-fixme[3]: Return type must be annotated.
+def _build_args(
+    pg: ProcessGroup, collective: str, example_tensor: torch.Tensor
+) -> List[Tuple[Any, ...]]:
     """
-    Helper function to test a set of collective operations on a given process group.
+    Helper function to build required args for a given collective.
     """
-
-    shape: torch.Size = example_tensor.shape
-    dtype: torch.dtype = example_tensor.dtype
-
-    # Create some dummy tensors for testing
     input_tensor = example_tensor.clone()
     output_tensors = [
         [torch.empty_like(input_tensor) for _ in range(get_world_size(pg))]
     ]
     tensor_list = [torch.empty_like(input_tensor)]
 
+    if collective == "allreduce":
+        return [
+            ([input_tensor], AllreduceOptions()),
+            ([input_tensor], ReduceOp.SUM),
+        ]
+    elif collective == "allgather":
+        return [(output_tensors, [input_tensor], AllgatherOptions())]
+    elif collective == "broadcast":
+        return [(tensor_list, BroadcastOptions())]
+    elif collective == "broadcast_one":
+        return [(input_tensor, 0)]
+    else:
+        raise ValueError(f"Unsupported collective: {collective}")
+
+
+def run_collective(
+    pg: ProcessGroup,
+    collective: str,
+    example_tensor: torch.Tensor = torch.randn((2, 3), dtype=torch.float32),
+) -> Dict[str, dist._Work]:
+    """Run a single collective."""
+    shape: torch.Size = example_tensor.shape
+    dtype: torch.dtype = example_tensor.dtype
+    coll = getattr(pg, collective)
+    args_list = _build_args(pg=pg, collective=collective, example_tensor=example_tensor)
```

Comment:
What's the intention behind pulling this out? I'm not really convinced that this makes it all that much cleaner

In some ways I think I'd prefer if we got rid of the arg generation and instead flatten this out with direct calls i.e.

```
if collective == "allreduce":
    work = pg.allreduce(...)

work.wait()
...
```